src/components/NewDatePicker/CalendarPicker/index.js

```diff
@@ -99,34 +99,34 @@ class CalendarPicker extends React.PureComponent {
     onDayPressed(day) {
         this.setState(
             (prev) => ({
-                currentDateView: moment(prev.currentDateView).set('date', day).toDate(),
+                currentDateView: setDate(new Date(prev.currentDateView), day),
             }),
-            () => this.props.onSelected(moment(this.state.currentDateView).format('YYYY-MM-DD')),
+            () => this.props.onSelected(format(new Date(this.state.currentDateView), CONST.DATE.FNS_FORMAT_STRING)),
         );
     }
 
     /**
      * Handles the user pressing the previous month arrow of the calendar picker.
      */
     moveToPrevMonth() {
-        this.setState((prev) => ({currentDateView: moment(prev.currentDateView).subtract(1, 'months').toDate()}));
+        this.setState((prev) => ({currentDateView: subMonths(new Date(prev.currentDateView), 1)}));
     }
 
     /**
      * Handles the user pressing the next month arrow of the calendar picker.
      */
     moveToNextMonth() {
-        this.setState((prev) => ({currentDateView: moment(prev.currentDateView).add(1, 'months').toDate()}));
+        this.setState((prev) => ({currentDateView: addMonths(new Date(prev.currentDateView), 1)}));
     }
 
     render() {
-        const monthNames = _.map(moment.localeData(this.props.preferredLocale).months(), Str.recapitalize);
-        const daysOfWeek = _.map(moment.localeData(this.props.preferredLocale).weekdays(), (day) => day.toUpperCase());
+        const monthNames = _.map(DateUtils.getMonthNames(this.props.preferredLocale), Str.recapitalize);
+        const daysOfWeek = _.map(DateUtils.getDaysOfWeek(this.props.preferredLocale), (day) => day.toUpperCase());
         const currentMonthView = this.state.currentDateView.getMonth();
         const currentYearView = this.state.currentDateView.getFullYear();
         const calendarDaysMatrix = generateMonthMatrix(currentYearView, currentMonthView);
-        const hasAvailableDatesNextMonth = moment(this.props.maxDate).endOf('month').endOf('day') >= moment(this.state.currentDateView).add(1, 'months');
-        const hasAvailableDatesPrevMonth = moment(this.props.minDate).startOf('month').startOf('day') <= moment(this.state.currentDateView).subtract(1, 'months');
+        const hasAvailableDatesNextMonth = startOfDay(endOfMonth(new Date(this.props.maxDate))) > addMonths(new Date(this.state.currentDateView), 1);
+        const hasAvailableDatesPrevMonth = startOfDay(new Date(this.props.minDate)) < endOfMonth(subMonths(new Date(this.state.currentDateView), 1));
```

Comment:
We should have the same logic here. This caused a regression https://github.com/Expensify/App/issues/31793